sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2230,6 +2230,8 @@ class Analyzer(
           val result = resolved transformDown {
             case UnresolvedMapObjects(func, inputData, cls) if inputData.resolved =>
               inputData.dataType match {
+                case ArrayType(et, false) if cls.isEmpty =>
```

Comment:
I agree with you. Now, I missed 2. Am I correct?

1. Already did by `cls.isEmpty`
2. We have to check whether `et` is primitive type or not at line 2233.
3. Already did by `ArrayType(et, false)`